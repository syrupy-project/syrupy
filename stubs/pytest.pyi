from typing import (
    Any,
    Callable,
    TypeVar,
)

ReturnType = TypeVar("ReturnType")

def fixture(func: Callable[..., ReturnType]) -> Callable[..., ReturnType]: ...

class Function: ...
class Item: ...
