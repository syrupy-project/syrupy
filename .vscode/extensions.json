{
    "recommendations": [
        "littlefoxteam.vscode-python-test-adapter",
        "ms-python.mypy-type-checker"
    ]
}
