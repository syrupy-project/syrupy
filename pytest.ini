[pytest]
addopts = -p pytester -p syrupy
