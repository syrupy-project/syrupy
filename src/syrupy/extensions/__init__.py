from .amber import AmberSnapshotExtension

DEFAULT_EXTENSION = AmberSnapshotExtension
