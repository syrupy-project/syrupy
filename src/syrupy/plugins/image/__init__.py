from .serializer import AbstractImageSnapshotSerializer


class PNGImageSnapshotSerializer(AbstractImageSnapshotSerializer):
    @property
    def extension(self) -> str:
        return "png"


class SVGImageSnapshotSerializer(AbstractImageSnapshotSerializer):
    @property
    def extension(self) -> str:
        return "svg"
