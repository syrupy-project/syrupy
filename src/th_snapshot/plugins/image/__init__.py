from .io import AbstractImageSnapshotIO
from .serializer import ImageSnapshotSerializer


class PNGImageSnapshotIO(AbstractImageSnapshotIO):
    extension = "png"


class SVGImageSnapshotIO(AbstractImageSnapshotIO):
    extension = "svg"
