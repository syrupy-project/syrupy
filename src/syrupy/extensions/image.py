from typing import (
    TYPE_CHECKING,
    Any,
)

from .single_file import SingleFileSnapshotExtension

if TYPE_CHECKING:
    from syrupy.types import SerializableData


class PNGImageSnapshotExtension(SingleFileSnapshotExtension):
    @property
    def _file_extension(self) -> str:
        return "png"


class SVGImageSnapshotExtension(SingleFileSnapshotExtension):
    @property
    def _file_extension(self) -> str:
        return "svg"

    def serialize(self, data: "SerializableData", **kwargs: Any) -> bytes:
        return str(data).encode("utf-8")
