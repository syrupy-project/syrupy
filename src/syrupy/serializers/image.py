from typing import TYPE_CHECKING

from .raw_single import RawSingleSnapshotSerializer


if TYPE_CHECKING:
    from syrupy.types import SerializableData


class PNGImageSnapshotSerializer(RawSingleSnapshotSerializer):
    @property
    def file_extension(self) -> str:
        return "png"

    def serialize(self, data: "SerializableData") -> bytes:
        if isinstance(data, bytes):
            return data
        raise TypeError(
            "Can not serialize image data. Expected "
            f"'bytes', got '{type(data).__name__}'."
        )


class SVGImageSnapshotSerializer(RawSingleSnapshotSerializer):
    @property
    def file_extension(self) -> str:
        return "svg"

    def serialize(self, data: "SerializableData") -> bytes:
        return str(data).encode("utf-8")
