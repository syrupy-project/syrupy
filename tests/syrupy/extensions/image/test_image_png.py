import base64

import pytest

from syrupy.extensions.image import PNGImageSnapshotExtension

actual_png = base64.b64decode(
    b"iVBORw0KGgoAAAANSUhEUgAAADIAAAAyBAMAAADsEZWCAAAAG1BMVEXMzMy"
    b"Wlpaqqqq3t7exsbGcnJy+vr6jo6PFxcUFpPI/AAAACXBIWXMAAA7EAAAOxA"
    b"GVKw4bAAAAQUlEQVQ4jWNgGAWjgP6ASdncAEaiAhaGiACmFhCJLsMaIiDAE"
    b"QEi0WXYEiMCOCJAJIY9KuYGTC0gknpuHwXDGwAA5fsIZw0iYWYAAAAASUVO"
    b"RK5CYII="
)


@pytest.fixture
def snapshot_png(snapshot):
    return snapshot.use_extension(PNGImageSnapshotExtension)


def test_image(snapshot_png):
    assert actual_png == snapshot_png


def test_multiple_snapshot_extensions(snapshot):
    """
    Example of switching extension classes on the fly.
    These should be indexed in order of assertion.
    """
    assert actual_png == snapshot(extension_class=PNGImageSnapshotExtension)
    assert actual_png == snapshot()  # uses initial extension class
    assert snapshot._extension is not None
    assert actual_png == snapshot(extension_class=PNGImageSnapshotExtension)
