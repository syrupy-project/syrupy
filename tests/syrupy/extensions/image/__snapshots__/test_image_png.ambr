# name: test_multiple_snapshot_extensions.1
  b'\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x002\x00\x00\x002\x04\x03\x00\x00\x00\xec\x11\x95\x82\x00\x00\x00\x1bPLTE\xcc\xcc\xcc\x96\x96\x96\xaa\xaa\xaa\xb7\xb7\xb7\xb1\xb1\xb1\x9c\x9c\x9c\xbe\xbe\xbe\xa3\xa3\xa3\xc5\xc5\xc5\x05\xa4\xf2?\x00\x00\x00\tpHYs\x00\x00\x0e\xc4\x00\x00\x0e\xc4\x01\x95+\x0e\x1b\x00\x00\x00AIDAT8\x8dc`\x18\x05\xa3\x80\xfe\x80I\xd9\xdc\x00F\xa2\x02\x16\x86\x88\x00\xa6\x16\x10\x89.\xc3\x1a" \xc0\x11\x01"\xd1e\xd8\x12#\x028"@$\x86=*\xe6\x06L- \x92zn\x1f\x05\xc3\x1b\x00\x00\xe5\xfb\x08g\r"af\x00\x00\x00\x00IEND\xaeB`\x82'
---
